aws/resource_aws_ec2_traffic_mirror_filter.go

```diff
@@ -0,0 +1,131 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsEc2TrafficMirrorFilter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEc2TrafficMirrorinFilterCreate,
+		Read:   resourceAwsEc2TrafficMirrorFilterRead,
+		Update: resourceAwsEc2TrafficMirrorFilterUpdate,
+		Delete: resourceAwsEc2TrafficMirrorFilterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"network_services": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"amazon-dns",
+					}, false),
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsEc2TrafficMirrorinFilterCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	input := &ec2.CreateTrafficMirrorFilterInput{}
+
+	if description, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(description.(string))
+	}
+
+	out, err := conn.CreateTrafficMirrorFilter(input)
+	if err != nil {
+		return fmt.Errorf("Error while creating traffic filter %s", err)
+	}
+
+	d.SetId(*out.TrafficMirrorFilter.TrafficMirrorFilterId)
+
+	return resourceAwsEc2TrafficMirrorFilterUpdate(d, meta)
+}
+
+func resourceAwsEc2TrafficMirrorFilterUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	if d.HasChange("network_services") {
+		input := &ec2.ModifyTrafficMirrorFilterNetworkServicesInput{
+			TrafficMirrorFilterId: aws.String(d.Id()),
+		}
+
+		o, n := d.GetChange("network_services")
+		newServices := n.(*schema.Set).Difference(o.(*schema.Set)).List()
+		if len(newServices) > 0 {
+			input.AddNetworkServices = expandStringList(newServices)
+		}
+
+		removeServices := o.(*schema.Set).Difference(n.(*schema.Set)).List()
+		if len(removeServices) > 0 {
+			input.RemoveNetworkServices = expandStringList(removeServices)
+		}
+
+		_, err := conn.ModifyTrafficMirrorFilterNetworkServices(input)
+		if err != nil {
+			return fmt.Errorf("Error modifying network services for traffic mirror filter %v", d.Id())
```

Comment:
This is not returning the API/SDK error for operators and code maintainers to troubleshoot the issue:

```suggestion
			return fmt.Errorf("error modifying EC2 Traffic Mirror Filter (%s) network services: %w", d.Id(), err)
```